lib/git-manager.js

```diff
@@ -411,6 +411,48 @@ function getCommit(repo, sha) {
    return githubRest.repos.getCommit(params({ ref: sha }, repo)).then(res => res.data);
 }
 
+function getCommits(repo, number) {
+   debug("Getting commits for pull #%s", number);
+   return github.paginate(githubRest.pulls.listCommits, params({ pull_number: number}, repo));
+}
+
+/**
+ * Get the lastest commit for determining signature validity
+ */
+function getLatestCommit(repo, number, head_sha) {
+   debug("Getting latest patch date for #%s", number);
+
+   // If we don't need to check for clean merges, just return the head commit
+   if (!config.ignoreCleanMerge) {
+      return getCommit(repo, head_sha);
+   }
+
+   // We only need to fetch the changes for merge commits
+   const shouldFetchFiles = commit => commit.parents.length > 1;
+   // Merge commits with no file changes are clean
+   const shouldIgnore = commit => shouldFetchFiles(commit) && !commit.files.length;
+
+   return getCommits(repo, number)
+      .then(commits => {
+         return Promise.all(commits.map(commit => {
+            if (shouldFetchFiles(commit)) {
+               return getCommit(repo, commit.sha);
+            } else {
+               return Promise.resolve(commit);
+            }
+         }))
+      })
+      .then(expandedCommits => {
```

Comment:
At this point, we have the all the commits for the whole pull, But the order seems unspecified. The API doesn't say anything about the order of commits, but I don't think there's anything we can intrinsically order them by. It seems like the rest of the code expects them to be in oldest-first order.

Reply:
I believe I read on a post somewhere that they were chronologically ordered by commit date by default, but it definitely does not guarantee it in the api documentation. I will add a change to explicitly check this.

[Here](https://github.blog/changelog/2020-07-14-pull-request-commits-now-ordered-chronologically/) is a blog post about it, but we shouldn't rely on that behavior.